pkg/login/social/social.go

```diff
@@ -96,12 +102,20 @@ func NewOAuthService() {
 			TlsClientKey:       sec.Key("tls_client_key").String(),
 			TlsClientCa:        sec.Key("tls_client_ca").String(),
 			TlsSkipVerify:      sec.Key("tls_skip_verify_insecure").MustBool(),
+			GroupMappingsFile:  sec.Key("group_mappings_file").String(),
 		}
 
 		if !info.Enabled {
 			continue
 		}
 
+		groupMappingsConfig, err := readConfig(info.GroupMappingsFile)
```

Comment:
Should only read config if group mapping file defined (`info.GroupMappingsFile != ""`). Right now checking out this code I get a panic on startup since returned `groupMappingsConfig` is nil and `info.GroupMappings = groupMappingsConfig.GroupMappings` panics - why I think it's easier to check if a file is provided here and only then read the config, handle error or assign group mappings.

But lets discuss this when my other comments/suggestions are fixed.

Reply:
I went ahead and made this change.